tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import logging
+from typing import Tuple
+
+from absl import flags
+from absl.testing import absltest
+
+from framework import xds_url_map_testcase
+from framework.test_app import client_app
+
+# Type aliases
+HostRule = xds_url_map_testcase.HostRule
+PathMatcher = xds_url_map_testcase.PathMatcher
+GcpResourceManager = xds_url_map_testcase.GcpResourceManager
+DumpedXdsConfig = xds_url_map_testcase.DumpedXdsConfig
+RpcTypeUnaryCall = xds_url_map_testcase.RpcTypeUnaryCall
+RpcTypeEmptyCall = xds_url_map_testcase.RpcTypeEmptyCall
+XdsTestClient = client_app.XdsTestClient
+
+logger = logging.getLogger(__name__)
+flags.adopt_module_key_flags(xds_url_map_testcase)
+
+_NUM_RPCS = 150
+match_labels = [{'name': 'TRAFFICDIRECTOR_NETWORK_NAME', 'value': 'default'}]
+not_match_labels = [{'name': 'fake', 'value': 'fail'}]
+
+
+class TestMetadataFilterMatchAll(xds_url_map_testcase.XdsUrlMapTestCase):
+
+    @staticmethod
+    def url_map_change(
+            host_rule: HostRule,
+            path_matcher: PathMatcher) -> Tuple[HostRule, PathMatcher]:
+        path_matcher["routeRules"] = [{
+            'priority': 0,
+            'matchRules': [{
+                'prefixMatch':
+                    '/',
+                'metadataFilters': [{
+                    'filterMatchCriteria': 'MATCH_ALL',
+                    'filterLabels': not_match_labels
+                }]
+            }],
+            'service': GcpResourceManager().default_backend_service()
+        }, {
+            'priority': 1,
+            'matchRules': [{
+                'prefixMatch':
+                    '/',
+                'metadataFilters': [{
+                    'filterMatchCriteria': 'MATCH_ALL',
+                    'filterLabels': match_labels
+                }]
+            }],
+            'service': GcpResourceManager().alternative_backend_service()
+        }]
+        return host_rule, path_matcher
+
+    def xds_config_validate(self, xds_config: DumpedXdsConfig):
+        self.assertNumEndpoints(xds_config, 2)
+        self.assertEqual(len(xds_config.rds['virtualHosts'][0]['routes']), 2)
+        logger.info('config:%s', xds_config)
```

Comment:
As discussed offline, let's try to be more concrete about if we are receiving the correct xDS config.

Reply:
Thanks Lidi. Updated.

Once approved we will run this before merge:
```
python3 -m tests.url_map --flagfile=config/grpc-testing.cfg --kube_context=gke_grpc-testing_us-central1-c_interop-test-psm-basic --client_image=gcr.io/grpc-testing/xds-interop/cpp-client:a069a322a9b3f49f0c2bdbd5ee363bfbab2471b0 --testing_version=master  --flagfile=config/url-map.cfg --strategy create
```
correct?